tests/test_api_materials.py

```
@@ -0,0 +1,754 @@
+    defaults.update(kwargs)
+    return MockRow(**defaults)
+
+
+# ---------------------------------------------------------------------------
+# Mock R2 bucket
+# ---------------------------------------------------------------------------
+
+class MockR2Bucket:
+    """Minimal R2 bucket stub that records calls."""
+
+    def __init__(self, presigned_url="https://r2.example.com/signed?token=abc"):
```

Ruff (0.16.1)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)